### PR TITLE
pythonPackages.boa-api: init at 0.1.14

### PR DESCRIPTION
**Pull request by swflint**
###### Description of changes

The boa-api package allows interfacing with the Boa infrastructure for MSR research

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/boa-api/default.nix [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+}:
+
+buildPythonPackage rec {
+  pname = "boa-api";
+  version = "0.1.13";
+
+  src = fetchFromGitHub {
+    owner = "boalang";
+    repo = "api-python";
+    rev = "8da603917e8df7a1b97be0eabf341efa8f8a7004";
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/boa-api/default.nix:13]:
```suggestion
    rev = "v${version}";
```

**Comment by SuperSandro2000** [comment]
@ofborg build python310Packages.boa-api

**Comment by SuperSandro2000** [comment]
fails to build:

```
util (unittest.loader._FailedTest) ... ERROR
======================================================================
ERROR: util (unittest.loader._FailedTest)
----------------------------------------------------------------------
ImportError: Failed to import test module: util
Traceback (most recent call last):
  File "/nix/store/ar76b3831ydkmfhf0r9v3wbma5i61iz2-python3-3.10.9/lib/python3.10/unittest/loader.py", line 154, in loadTestsFromName
    module = __import__(module_name)
  File "/build/source/boaapi/util.py", line 26, in <module>
    import boaapi.boa_client
  File "/build/source/boaapi/boa_client.py", line 20, in <module>
    from boaapi.util import CookiesTransport, parse_job, fetch_url
ImportError: cannot import name 'CookiesTransport' from partially initialized module 'boaapi.util' (most likely due to a circular import) (/build/source/boaapi/util.py)
----------------------------------------------------------------------
Ran 1 test in 0.000s
FAILED (errors=1)
Test failed: <unittest.runner.TextTestResult run=1 errors=1 failures=0>
```

**Comment by swflint** [comment]
Talked to upstream.  They cut a new release with a fix.

**Comment by SuperSandro2000** [comment]
@ofborg build python310Packages.boa-api

**Pull request merged**